objectherkenning_openbare_ruimte/databricks_pipelines/common/table_manager.py

```diff
@@ -0,0 +1,182 @@
+        pending_table_rows = table_rows.filter("status = 'Pending'")
+        print(
+            f"Filtered to {pending_table_rows.count()} 'Pending' rows from {self.catalog}.{self.schema}.{table_name}."
+        )
+        return pending_table_rows
+
+    def load_from_table(self, table_name: str) -> DataFrame:
+        """
+        Loads all rows from the specified table in the catalog and schema.
+
+        Parameters:
+        ----------
+        table_name: The name of the table to load.
+
+        Returns:
+        -------
+        DataFrame: A DataFrame containing all rows from the specified table.
+        """
+        return self._load_table(table_name)
+
+    def remove_fields_from_table_schema(
+        self, table_name: str, fields_to_remove: set
+    ) -> StructType:
+        """
+        This method loads the schema of the specified table, removes the fields
+        listed in `fields_to_remove`, and returns the modified schema.
+
+        Parameters:
+        ----------
+        table_name: The name of the table whose schema will be modified.
+        fields_to_remove: A set of field names to be removed from the schema.
+
+        Returns:
+        -------
+        StructType: The modified schema with the specified fields removed.
+
+        """
+        table_schema = self._load_table(table_name=table_name).schema
+
+        # Modify the schema by removing the specified fields
+        modified_schema = StructType(
+            [field for field in table_schema if field.name not in fields_to_remove]
+        )
+
+        return modified_schema
+
+    # def update_status(
+    #     self, table_name: str, job_process_time: datetime, exclude_ids=[]
+    # ):
+    #     table_df = self.spark.table(f"{self.catalog}.{self.schema}.{table_name}")
+
+    #     pending_df = table_df.filter(
+    #         (col("status") == "Pending") & (~col("id").isin(exclude_ids))
+    #     )
+
+    #     total_pending_before = pending_df.count()
+
+    #     updated_df = pending_df.withColumn("status", lit("Processed")).withColumn(
+    #         "processed_at", lit(job_process_time)
+    #     )
+
+    #     updated_df.write.mode("overwrite").insertInto(
+    #         f"{self.catalog}.{self.schema}.{table_name}"
+    #     )
+
+    #     table_df = self.spark.table(f"{self.catalog}.{self.schema}.{table_name}")
+    #     total_pending_after = table_df.filter(col("status") == "Pending").count()
+
+    #     updated_rows = total_pending_before - total_pending_after
+
+    #     print(
+    #         f"Updated {updated_rows} 'Pending' rows to 'Processed' in {self.catalog}.{self.schema}.{table_name}, {total_pending_after} rows remained 'Pending'."
+    #     )
+
+    def write_to_table(self, df, table_name, mode="append"):
+        df.write.mode(mode).saveAsTable(f"{self.catalog}.{self.schema}.{table_name}")
+        print(f"Appended {df.count()} rows to {table_name}.")
+
+    @staticmethod
+    def compare_dataframes(df1, df2, df1_name, df2_name):
```

Comment:
This doesn't look it should be on table manager since it's more a generic dataframe function.